[Bug]: Large Lidarr installations cause string overflow

### Bug Description

When getReconciliationSnapshot() is calling the lidarr API to retrieve all albums:
[`                this.client.get("/api/v1/album").catch((err) => {
                    logger.error("[LIDARR] Failed to fetch albums for snapshot:", err.message);
                    return { data: [] };
                }),`](https://github.com/Chevron7Locked/kima-hub/blob/376ae0590a0170c716a2fb02ad81f9202529c446/backend/src/services/lidarr.ts#L2418C1-L2421C20)

The call errors for large Lidarr installations - albums overflows. 

### Steps to Reproduce

1. Lidarr configured with large number of artists and albums (In my case the album api endpoint returns 650MB of data)
2. Error is produced

### Expected Behavior

Lidarr snapshot returns all data

### Actual Behavior

Snapshot failed with:
`[ERROR] [LIDARR] Failed to fetch albums for snapshot: Cannot create a string longer than 0x1fffffe8 characters`

### Kima Version

v.1.5.10

### Deployment Method

Docker (docker-compose)

### Environment Details

_No response_

### Relevant Logs

```shell
[ERROR] [LIDARR] Failed to fetch albums for snapshot: Cannot create a string longer than 0x1fffffe8 characters
[ERROR] [LIDARR] Failed to fetch albums for snapshot: Cannot create a string longer than 0x1fffffe8 characters
[ERROR] [LIDARR] Failed to fetch albums for snapshot: Cannot create a string longer than 0x1fffffe8 characters
```

### Checklist

- [x] I have searched existing issues to ensure this bug hasn't already been reported
- [x] I am using a supported version of Kima

--- a/backend/src/services/lidarr.ts
+++ b/backend/src/services/lidarr.ts
@@ -2407,17 +2407,58 @@ class LidarrService {
         }
 
         try {
+            const fetchAlbumsForSnapshot = async (): Promise<LidarrAlbum[]> => {
+                const albums: LidarrAlbum[] = [];
+
+                const artistsResponse = await this.client!.get("/api/v1/artist");
+                const artists = Array.isArray(artistsResponse.data)
+                    ? (artistsResponse.data as LidarrArtist[])
+                    : [];
+
+                const artistBatchSize = 25;
+                for (let start = 0; start < artists.length; start += artistBatchSize) {
+                    const batch = artists.slice(start, start + artistBatchSize);
+                    const batchResponses = await Promise.all(
+                        batch.map((artist) =>
+                            this.client!
+                                .get("/api/v1/album", {
+                                    params: { artistId: artist.id },
+                                })
+                                .catch((err: any) => {
+                                    logger.warn(
+                                        `[LIDARR] Failed to fetch albums for artist ${artist.id} during snapshot:`,
+                                        err.message
+                                    );
+                                    return { data: [] };
+                                })
+                        )
+                    );
+
+                    for (const response of batchResponses) {
+                        if (Array.isArray(response.data)) {
+                            albums.push(...(response.data as LidarrAlbum[]));
+                        }
+                    }
+                }
+
+                logger.debug(
+                    `[LIDARR] Snapshot album fetch: ${artists.length} artists, ${albums.length} albums fetched`
+                );
+
+                return albums;
+            };
+
             // Fetch queue and albums in parallel
             const [queueResponse, albumsResponse] = await Promise.all([
                 this.client.get("/api/v1/queue", {
                     params: { page: 1, pageSize: 1000, includeUnknownArtistItems: true },
-                }).catch((err) => {
+                }).catch((err: any) => {
                     logger.error("[LIDARR] Failed to fetch queue for snapshot:", err.message);
                     return { data: { records: [] } };
                 }),
-                this.client.get("/api/v1/album").catch((err) => {
+                fetchAlbumsForSnapshot().catch((err: any) => {
                     logger.error("[LIDARR] Failed to fetch albums for snapshot:", err.message);
-                    return { data: [] };
+                    return [];
                 }),
             ]);
 
@@ -2438,7 +2479,7 @@ class LidarrService {
             }
 
             // Index albums by MBID and normalized title
-            const albums = albumsResponse.data || [];
+            const albums = albumsResponse || [];
             for (const album of albums) {
                 const hasFiles = album.statistics?.percentOfTracks > 0;
                 if (!hasFiles) continue; // Only index albums with files

